pkg/search/search.go

```diff
@@ -0,0 +1,88 @@
+package search
+
+import (
+	"encoding/json"
+	//"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/maxott/magda-cli/pkg/adapter"
+	log "go.uber.org/zap"
+)
+
+/**** DATASETS ****/
+
+type DatasetRequest struct {
+	Query     string
+	Offset    int
+	Limit     int
+	Publisher string
+}
+
+type DatasetResult struct {
+	HitCount       int   `json:"hitCount"`
+	DataSets       []struct {
+		Title					string	`json:"title"`
+		Description				string	`json:"description"`
+		Issued					string `json:"issued"`
+		Modified				string `json:"modified"`
+		Languages				[]string `json:"languages"`
+		Publisher				string `json:"publisher"`
+		AccrualPeriodicity			string `json:"accrualPeriodicity"`
+		AccrualPeriodicityRecurrenceRule	string `json:"accrualPeriodicityRecurrenceRule"`
+		Themes					[]string `json:"themes"`
+		Keywords				[]string `json:"keywords"`
+		ContactPoint				string `json:"contactPoint"`
+		LandingPage				string `json:"landingPage"`
+		DefaultLicense				string `json:"defaultLicense"`
+	} `json:"dataSets"`
+}
+
+func Dataset(cmd *DatasetRequest, adpt *adapter.Adapter, logger *log.Logger) (DatasetResult, error) {
+	pyl, err := DatasetRaw(cmd, adpt, logger)
+	if err != nil {
+		return DatasetResult{}, err
+	}
+	res := DatasetResult{}
+	_ = json.Unmarshal(pyl.AsBytes(), &res)
+	return res, nil
+}
+
+func DatasetRaw(cmd *DatasetRequest, adpt *adapter.Adapter, logger *log.Logger) (adapter.Payload, error) {
+	path := recordPath(nil, adpt)
+
+	q := []string{}
+
+	if cmd.Query != "" {
```

Comment:
Does it still make sense to make this call without a 'query'? Maybe if 'Publisher' is set? What happens if both are empty?

Reply:
As far as I'm aware, this endpoint is designed for use with query. If publisher only is used it applies over all datasets, and if none are used it just returns all datasets (respecting the limit).

I would rather leave this as-is and not include any checks here, that way this CLI remains a thin client to the API. Magda itself has no such checks as of now, what do you reckon?

Try this https://dev.magda.io/api/v0/search/datasets?publisher=test